install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml

```diff
@@ -148,6 +148,38 @@ spec:
         {{- if $.Values.global.proxy.envoyAccessLogService.enabled }}
           - --envoyAccessLogServiceAddress
           - {{ $.Values.global.proxy.envoyAccessLogService.host }}:{{ $.Values.global.proxy.envoyAccessLogService.port }}
+        {{- if $.Values.global.proxy.envoyAccessLogService.tlsSettings }}
+          - --envoyAccessLogServiceTlsMode
+          - {{ $.Values.global.proxy.envoyAccessLogService.tlsSettings.mode }}
+        {{- if $.Values.global.proxy.envoyAccessLogService.tlsSettings.clientCertificate }}
+          - --envoyAccessLogServiceTlsClientCertificate
+          - {{ $.Values.global.proxy.envoyAccessLogService.tlsSettings.clientCertificate }}
+        {{- end }}
+        {{- if $.Values.global.proxy.envoyAccessLogService.tlsSettings.privateKey }}
+          - --envoyAccessLogServiceTlsPrivateKey
+          - {{ $.Values.global.proxy.envoyAccessLogService.tlsSettings.privateKey }}
+        {{- end }}
+        {{- if $.Values.global.proxy.envoyAccessLogService.tlsSettings.caCertificates }}
+          - --envoyAccessLogServiceTlsCaCertificates
+          - {{ $.Values.global.proxy.envoyAccessLogService.tlsSettings.caCertificates }}
+        {{- end }}
+        {{- if $.Values.global.proxy.envoyAccessLogService.tlsSettings.sni }}
+          - --envoyAccessLogServiceTlsSni
+          - {{ $.Values.global.proxy.envoyAccessLogService.tlsSettings.sni }}
+        {{- end }}
+        {{- if $.Values.global.proxy.envoyAccessLogService.tlsSettings.subjectAltNames }}
+          - --envoyAccessLogServiceTlsSubjectAltNames
+          - {{ join "," $.Values.global.proxy.envoyAccessLogService.tlsSettings.subjectAltNames | quote }}
+        {{- end }}
+        {{- end }}
+        {{- if $.Values.global.proxy.envoyAccessLogService.tcpKeepalive }}
+          - --envoyAccessLogServiceTcpKeepaliveProbes
+          - {{ $.Values.global.proxy.envoyAccessLogService.tcpKeepalive.probes }}
+          - --envoyAccessLogServiceTcpKeepaliveTime
+          - {{ $.Values.global.proxy.envoyAccessLogService.tcpKeepalive.time }}
+          - --envoyAccessLogServiceTcpKeepaliveInterval
+          - {{ $.Values.global.proxy.envoyAccessLogService.tcpKeepalive.interval }}
+        {{- end }}
```

Comment:
woa.. this is a lot of options. Isn't there a more compact way of specifying this ?

Reply:
My idea is to do this like `--envoyAccessLogServiceTLS MUTUAL:/etc/istio/als/cert-chain.pem:/etc/istio/als/key.pem...`. So we can compress them to three options.

Reply:
@howardjohn / @mandarjog ideas?

Reply:
could you maybe instead use proto text or similar to supply the `tls_context` for the cluster?

`--envoyAccessLogServiceTLSContext={common_tls_context: [{certificate_chain: { filename: /etc/istio/als/cert-chain.pem} },], sni: "<foo>"}`

or similar?

That would avoid creating a custom compression for the options -- and is probably more future looking (when these kinds of configuration are providing in Sidecar API).

Reply:
@douglas-reid thanks for your advice, it makes much sense to me. I'm going to refactor the codes with this way.